### ISSUE TITLE
GPX files from Suunto and TomTom not recognised by Laufhelden

### ISSUE THREAD

**Issue by ncartron** [jdrescher2006/Laufhelden#149]
Hi,

in my effort to get rid of my non-SailfishOS phones, I am trying not to use Android or iOS to sync with Strava, and therefore wants to use Laufhelden to upload my activities to Strava.

It works well when running with Laufhelden, or with an Amazfit Bip (thanks to Amazfish app), however when download a GPX file from Suunto (via their Movescount.com website) or Tomtom, the GPX file is not recognised by Laufhelden and is just displayed as "0 kms".

I did some quick "debugging" and found out that by removing the "<name>activity</name>"  which follows the "<trk>" section, it works well and Laufhelden recognises the GPX file (displays map, charts and info).

I have added the below  screenshot to explain, but basically removing this line fixes the issue:

![screenshot_2019-01-09_15-55-02](https://user-images.githubusercontent.com/22927685/50907163-ff891f00-1426-11e9-9151-adf575757d02.png)

**Comment by jdrescher2006** [comment]
Hi Nico,
can you send that GPX file to me? I would try and make Laufhelden compatible to it.

**Comment by ncartron** [comment]
I sent you a message using the contact form on Openrepos - i don't have your email address :)

**Comment by jdrescher2006** [comment]
Interesting file, thanks for that. It is good to make Laufhelden compatible to that type of file.
It would be interesting how the keywords (<name>RUNNING</name>) are for other workout types in that file type. Maybe you have a few examples?

**Comment by jdrescher2006** [comment]
That would be good. That way we could raise compatibility to this format.

**Comment by jdrescher2006** [comment]
The file is basically a GPX standard file. With only one difference and that is the highlighted line in the posted image.
I do not think we need a new parser for this file, we just have to tweak the old parser and everything will be fine. I am working on that right now.

**Comment by berndhahnebach** [comment]
any news on this?

**Comment by martonmiklos** [comment]
> any news on this?

Not really. Jens abandoned the project, and there were not much development since then. If you could post an example GPX I might be able to look into it. 

Edit: By looking into the actual code and the screenshot in the original issue described by @ncartron it should just work.
The parser looks for the trk tag, and reads the childs of that until the trkpt or trkseg is not found. The tags with other keys are simply ignored.

**Comment by hjdfdr44** [comment]
What is the long time plan for those? I suppose it might be useful to host them in the equal repo to start [build pickleball court](https://pickleballmate.com/build-a-pickleball-court/) where the localized asset rpms spec document lives.

**Comment by berndhahnebach** [comment]
I just wanted to know about this because this could be a game changer if it does not work. But as you wrote it should work.
Offtopic ...
Does it still run on latest Sailfish 4.5.x ?
Is it worth giving it a try?

cheers bernd

**Comment by martonmiklos** [comment]
Hi @berndhahnebach 
Is Sailfish 4.5.x released already?

**Comment by berndhahnebach** [comment]
upps 4.4. ... 
https://forum.sailfishos.org/tag/release-notes

**Comment by martonmiklos** [comment]
>  Is it worth giving it a try?

I used to use it on my XA2, however the GPS lock time got irregularly long thanks to some software bug in the system itself (not in Laufhelden), and I have not used it recently. Takes ages to get GPS fix. But give it a try an report back.
